contrib/mesos/pkg/queue/delay/delay_queue.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package delay
+
+import (
+	"container/heap"
+	"sync"
+
+	"github.com/pivotal-golang/clock"
+
+	"k8s.io/kubernetes/contrib/mesos/pkg/queue/priority"
+)
+
+// Queue is a thread-safe, time-based queue.
+//
+// Items can only be popped after their event time has been reached.
+// Use Add to specify a delay duration and have the event time calculated.
+// Use Offer to specify a specific event time.
+type Queue struct {
+	*priority.Queue
+	lock  *sync.RWMutex
+	cond  *sync.Cond
+	clock clock.Clock
+}
+
+func NewDelayQueue(clock clock.Clock) *Queue {
```

Comment:
I was using delay.NewQueue for a while, but delay.NewDelayQueue seemed to distinguish it well from delay.NewIndexedDelayQueue and priority.NewQueue.

So I'm on the fence...